scipy/stats/_qmc.py

```diff
@@ -0,0 +1,1587 @@
+       Technometrics, 1979.
+    .. [2] M. Stein, "Large sample properties of simulations using Latin
+       hypercube sampling." Technometrics 29, no. 2: 143-151, 1987.
+    .. [3] A. B. Owen, "Monte Carlo variance of scrambled net quadrature."
+       SIAM Journal on Numerical Analysis 34, no. 5: 1884-1910, 1997
+    .. [4]  Loh, W.-L. "On Latin hypercube sampling." The annals of statistics
+       24, no. 5: 2058-2080, 1996.
+
+    Examples
+    --------
+    Generate samples from a Latin hypercube generator.
+
+    >>> from scipy.stats import qmc
+    >>> sampler = qmc.LatinHypercube(d=2, seed=12345)
+    >>> sample = sampler.random(n=5)
+    >>> sample
+    array([[0.01407678, 0.53672489],
+           [0.36321624, 0.75908794],
+           [0.28645499, 0.48089106],
+           [0.2070971 , 0.46936458],
+           [0.45021867, 0.66928603]])
+
+    Compute the quality of the sample using the discrepancy criterion.
+
+    >>> qmc.discrepancy(sample)
+    0.1271335273223828
+
+    Finally, samples can be scaled to bounds.
+
+    >>> bounds = [[0, 2], [10, 5]]
+    >>> qmc.scale(sample, bounds)
+    array([[0.14076781, 3.61017467],
+           [3.63216238, 4.27726383],
+           [2.86454994, 3.44267318],
+           [2.07097096, 3.40809374],
+           [4.50218672, 4.00785808]])
+
+    """
+
+    def __init__(self, d: int, centered: bool = False,
+                 seed: Optional[int, np.random.RandomState] = None):
+        super().__init__(d=d, seed=seed)
+        self.centered = centered
+
+    def random(self, n: int = 1) -> np.ndarray:
+        """Draw `n` in the half-open interval ``[0, 1)``.
+
+        Parameters
+        ----------
+        n : int
+            Number of samples to generate in the parameter space. Default is 1.
+
+        Returns
+        -------
+        sample : array_like (n, d)
+            LHS sample.
+
+        """
+        if self.centered:
+            r = 0.5
+        else:
+            r = self.rng.random_sample((n, self.d))
+
+        try:
+            rg_integers = self.rng.integers
+        except AttributeError:
+            rg_integers = self.rng.randint
+        q = rg_integers(low=1, high=n, size=(n, self.d))
+
+        return 1. / n * (q - r)
+
+
+class OptimalDesign(QMCEngine):
+    """Optimal design.
+
+    Optimize the design by doing random permutations to lower the centered
+    discrepancy. If `optimization` is False, `niter` design are generated and
+    the one with lowest centered discrepancy is return. This option is faster.
+    Otherwise,
+
+    Centered discrepancy-based design shows better space filling robustness
+    toward 2D and 3D subprojections. Distance-based design shows better space
+    filling but less robustness to subprojections.
+
+    Parameters
+    ----------
+    d : int
+        Dimension of the parameter space.
+    start_design : array_like (n, d), optional
+        Initial design of experiment to optimize.
+    niter : int
```

Comment:
`, optional` missing for this and the next parameter descriptions

Reply:
Why is this optional? I thought this keyword was only for params defaulting to None?

Reply:
No, optional in parameter descriptions simply means "you don't have to specify this when calling the function". It long predates the `Optional` in type annotations, which indeed are shorthand for `Union[None, ...]`.

Reply:
Oh ok I see, I will update all missing optional then.